### PR TITLE
Handle Missing Project or Dataset by Displaying Error Message Instead…

### PR DESCRIPTION
**Pull request by hassaanfarooq01**
This PR ensures that an error message is displayed and empty response for a project and dataset if their IDs do not exist, instead of raising an exception.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced error handling in dataset and project data retrieval methods. 🛠️

### 📊 Key Changes
- Added checks for: 
  - The presence of a dataset or project ID before proceeding 👤
  - Successful response from the server before processing 🌐
  - The response object having a `.json()` method to ensure valid data format 📄
  - The existence of actual data within the response to ensure meaningful data acquisition 📊
- Implemented robust error handling to gracefully handle failures at any step with informative logging 🚨

### 🎯 Purpose & Impact
- **Purpose**: 
  - To make data retrieval methods more reliable and informative when failures occur. This aims to improve debugging and user experience by providing clear error messages.
- **Impact**: 
  - Users will experience fewer crashes and more descriptive error messages, making it easier to understand and resolve issues. This change enhances the robustness of applications built on the `hub-sdk`.
  - Developers can more easily diagnose problems thanks to detailed logging. This ultimately contributes to a smoother development process and more stable applications.

### CODE REVIEW

**Review by sergiuwaxmann — APPROVED** [review]

**Comment by sergiuwaxmann** [comment]
Failed tests on this PR are fixed [here](https://github.com/ultralytics/hub-sdk/pull/129).

File: hub_sdk/modules/projects.py [hunk 1]
```diff
@@ -49,12 +49,32 @@ def get_data(self) -> None:
...
+            resp_data = response.json()
+            if resp_data is None:
+                self.logger.error("No data received in the response for project id %s", self.id)
+                return
+
+            self.data = resp_data.get("data", {})
+            self.logger.debug("Project data retrieved for id %s", self.id)
+
+        except Exception as e:
+            self.logger.error("An error occurred while retrieving data for project id %s: %s", self.id, str(e))
```
Comment (glenn-jocher) [inline hub_sdk/modules/projects.py:77]: @hassaanfarooq01 we should use fstrings in all places now, the percentage sign usage is legacy and should not be used for new code, i.e.:

```diff
self.logger.error("An error occurred while retrieving data for project id %s: %s", self.id, str(e))
self.logger.error(f"An error occurred while retrieving data for project id {self.id}: {e}")
```

Can you apply fstrings to the 10 logger lines in the PR? Then it should be good to go. Thanks!

Comment (hassaanfarooq01) [reply hub_sdk/modules/projects.py:77]: @glenn-jocher Yeah Sure

**Comment by UltralyticsAssistant** [comment]
@hassaanfarooq01 @yogendrasinghx tests are passing. Good to merge this PR now?

**Comment by UltralyticsAssistant** [comment]
PR merged!

**Pull request merged**